modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/layout-data-items/ColumnWithControls.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import React from 'react';
+
+import {LAYOUT_DATA_ITEM_TYPES} from '../../config/constants/layoutDataItemTypes';
+import Topper from '../Topper';
+import Column from './Column';
+
+const ColumnWithControls = React.forwardRef(
+	({children, item, layoutData}, ref) => {
+		return (
+			<Topper
+				acceptDrop={[
+					LAYOUT_DATA_ITEM_TYPES.fragment,
+					LAYOUT_DATA_ITEM_TYPES.row
+				]}
+				active={false}
+				item={item}
+				layoutData={layoutData}
+				name={Liferay.Language.get('column')}
+			>
+				{() => (
+					<Column className="page-editor__col" item={item} ref={ref}>
+						{children}
+					</Column>
+				)}
+			</Topper>
+		);
+	}
+);
+
+export default ColumnWithControls;
```

Comment:
Hey @victorg1991 Just a suggestion to minimize these respective files for each layoutData, we can create a simpler abstraction, for example:

`withControls.js`
```jsx
export const withControls = WrapperComponent => React.forwardRef(({item, ...props}, ref) => (
	<Topper item={item} {...props}>
		<WrapperComponent item={item} ref={ref} />
	</Topper>
));
```

`PageEditor.js`
```jsx
const LAYOUT_DATA_ITEMS = {
	[LAYOUT_DATA_ITEM_TYPES.column]: withControls(Column),
	[LAYOUT_DATA_ITEM_TYPES.container]: withControls(Container),
	[LAYOUT_DATA_ITEM_TYPES.fragment]: withControls(Fragment),
	[LAYOUT_DATA_ITEM_TYPES.root]: Root,
	[LAYOUT_DATA_ITEM_TYPES.row]: withControls(Row)
};
```

Reply:
I didn't want to make a lot of abstractions until things settle a bit, because for example now everything has a topper, which makes no sense, but it is like this because it controls the drag and drop, and things like that, We can come back an implement this later :)

Reply:
thanks!

Reply:
No problem.